### ISSUE TITLE
“<method 'append' of 'list' objects> returned a result with an exception set”

### ISSUE THREAD

**Issue by GluttonousCat** [abetlen/llama-cpp-python#1006]
I encountered this error when using the embedding function "llama.create_embedding", under this code. abstract is a text about 700 tokens, when i use the previous version 0.1.83, it can run. now the version is 0.2.22 
```
llama = Llama(model_path='./llama-2-7b.Q4_K_M.gguf', embedding=True, n_ctx=2048, n_gpu_layers=30)
abstract_embedding = llama.create_embedding(input=abstract).get('data')[0].get('embedding')
```
this is the report
```
Traceback (most recent call last):
  File "C:\Users\AI\PycharmProjects\roberta-gat\demo.py", line 107, in <module>
    get_abstract_embedding(path='data/validation.csv', start=0)
  File "C:\Users\AI\PycharmProjects\roberta-gat\demo.py", line 50, in get_abstract_embedding
    abstract_embedding = llama.create_embedding(input=abstract).get('data')[0].get('embedding')
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\AI\.conda\envs\roberta-gat\Lib\site-packages\llama_cpp\llama.py", line 1309, in create_embedding
    data.append(
SystemError: <method 'append' of 'list' objects> returned a result with an exception set
```
This is the source code , and I find the value of llama_cpp.llama_get_embeddings(self._ctx.ctx) maybe is null?
```
for index, input in enumerate(inputs):     tokens = self.tokenize(input.encode("utf-8"), special=True)     self.reset()     self.eval(tokens)     n_tokens = len(tokens)     total_tokens += n_tokens     embedding = llama_cpp.llama_get_embeddings(self._ctx.ctx)[         : llama_cpp.llama_n_embd(self._model.model)     ]      data.append(         {             "object": "embedding",             "embedding": embedding,             "index": index,         }     )
```

**Comment by CharlieChenDSS** [comment]
got the same error with llama-cpp-python v0.2.22 : 

```
  File "/Users/xxxx/opt/anaconda3/envs/langchain_python_v3.9/lib/python3.9/site-packages/llama_cpp/llama.py", line 1309, in create_embedding
    data.append(
SystemError: <method 'append' of 'list' objects> returned a result with an error set

```

**Comment by CharlieChenDSS** [comment]
@abetlen could you please help look into this issue when you get a chance? Thanks. 
It could be raised from below code block, function `create_embedding` of [llama.py](https://github.com/abetlen/llama-cpp-python/blob/main/llama_cpp/llama.py)
`embedding = llama_cpp.llama_get_embeddings(self._ctx.ctx)[
                : llama_cpp.llama_n_embd(self._model.model)
            ]`

**Comment by GaryZhen** [comment]
I meet that problem as well. I am using Chroma, which asks me to specify the embedding function.
I have to use llama-cpp-embedding from langchain. However, it always shows this problem.

**Comment by brandonrobertz** [comment]
I was able to replicate this and was able to fix on my end. Does this PR also fix for y'all? https://github.com/abetlen/llama-cpp-python/pull/1019

**Comment by zakki** [comment]
Thank you! I had same problem and 1019 fixed that.

**Comment by GluttonousCat** [comment]
> I was able to replicate this and was able to fix on my end. Does this PR also fix for y'all? #1019

Thank you! i fixed that error by #1019
